java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -442,9 +442,9 @@ abstract class AbstractArcHost(
             stopParticle(particleContext)
         }
         maybeCancelResurrection(context)
-        context.entityHandleManager.close()
         context.arcState = ArcState.Stopped
         updateArcHostContext(arcId, context)
+        context.entityHandleManager.close()
```

Comment:
In general, this still might close the service connection abruptly while work is being done? Should we make that wait some time for idle()?

Reply:
Good question, I'm not sure. 

Calling `close` won't stop the work that the Store is doing, it just means that the StorageProxy and Handles will no longer receive callbacks related to changes. The expectation is that `close` is called once all `Handles` are no longer expected to perform any actions, so this seems OK from that angle.

`idle` isn't exposed in `StorageCommunicationEndpoint`, so we'd either have to expose it, or adjust Store implementations to `idle` at appropriate times (for example, having `close` calls block on `idle` first). It looks like `idle` isn't implemented at all for `ServiceStore`, at the moment.

It probably makes sense to wait until @jasonwyatt has landed the threading model changes, and then tackle this problem holistically.